privacyidea/lib/smsprovider/HttpSMSProvider.py

```diff
@@ -46,13 +48,26 @@
 from privacyidea.lib import _
 import requests
 from urlparse import urlparse
-
+import re
 import logging
 log = logging.getLogger(__name__)
 
 
 class HttpSMSProvider(ISMSProvider):
 
+    def _mangle_phone(self, phone, config):
+        regexp = config.get("REGEXP")
+        if regexp:
+            try:
+                m = re.match("^/(.*)/(.*)/$", regexp)
+                if m:
+                    phone = re.sub(m.group(1), m.group(2), phone)
+            except re.error:
+                log.warning("Can not mangle phone number. "
+                            "Please check your REGEXP: {0!s}".format(regexp))
```

Comment:
Small nitpick: We should either use `u""` or `{0!r}` here to avoid errors if `regexp` contains non-ASCII characters

Reply:
Yup, I think using ``u""`` would be best!

Reply:
done